<!--
  Waiterbot remote app:
  - 3d sensor
  - ir half ring
  - virtual sensor (throat-cutting tables)
  - safety controller
  - auto-docking
  - ar marker tracker
  - navigation manager
  - move_base
  - amcl localization
  
  Before starting the app, the bootstrap setting must be running in the robot:
    $ roslaunch turtlebot_bringup minimal.launch rapp_lists:=cafe_rapps/cafe.rapps
  The cafe rapps contains (between others) the waiterbot app.
 -->

<launch>
  <!-- Initial position -->
  <arg name="initial_pose_x" value="2.0"/>
  <arg name="initial_pose_y" value="2.0"/>
  <arg name="initial_pose_a" value="0.0"/>

  <!--  ****************** 3d sensor ******************  -->
  <include file="$(find turtlebot_bringup)/launch/3dsensor.launch">
    <arg name="rgb_processing" value="true"/>
    <arg name="scan_processing" value="true"/>
    <arg name="depth_processing" value="false"/>
  </include>

  <!--  ***** Kobuki mobile base safety controller ****  -->
  <include file="$(find sniperbot)/launch/includes/_safety_ctrl.launch"/>

  <!-- Reconfigure command velocity multiplexer to fit waiterbot needs. Just set the parameter doesn't work
       if we start the waiterbot rapp with app manager; we must explicitly call the reconfigure service -->
  <node pkg="rosservice" type="rosservice" name="reconfig_vel_mux" args="call --wait /cmd_vel_mux/set_parameters
            '{config:{strs:[[yaml_cfg_file,/home/turtlebot/cafe_concert/src/kobuki-x/waiterbot/param/vel_multiplexer.yaml]]}}'"/>

  <!--  ******** Kobuki auto-docking controller *******  -->
  <include file="$(find sniperbot)/launch/includes/_autodock.launch"/>

  <!--  ********** Alvar AR markers tracker ***********  -->
  <include file="$(find sniperbot)/launch/includes/_ar_tracker.launch">
    <arg name="max_frequency" value="4.0"/>  <!-- This is a CPU-hungry bastard; reduce its rate --> 
  </include>

  <!--  ************* Navigation manager **************  -->
  <include file="$(find sniperbot)/launch/includes/_master_ctrl.launch"/>

  <!--  ************** Rosnav move base ***************  -->
  <include file="$(find sniperbot)/launch/includes/_move_base.launch"/>

  <!--  ******************** Amcl *********************  -->
  <include file="$(find sniperbot)/launch/includes/_amcl.launch">
    <arg name="scan_topic" value="/scan"/>
    <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
    <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
    <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
  </include>
</launch>
